usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClient.java

```diff
@@ -0,0 +1,202 @@
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + SLASH + petName,
+                accessToken).get();
+        List<Weight> weightList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] weights = jsonArray.split(BRACKET_SLASH);
+            weightList.add(GSON.fromJson(weights[0], Weight.class));
+            for (int i = 1; i < weights.length; i++) {
+                weights[i] = BRACKET + weights[i];
+                weightList.add(GSON.fromJson(weights[i], Weight.class));
+            }
+        }
+        return weightList;
+    }
+
+    /**
+     * Gets the data from all the weights of the pet between the initial and final
+     * date not including them.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the weights eaten by the pet in the specified time
+     */
+    public List<Weight> getAllWeightsBetween(String accessToken, String owner, String petName,
+                                             DateTime initialDate, DateTime finalDate)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + SLASH + petName + "/between/"
+                        + initialDate + SLASH + finalDate, accessToken).get();
+        List<Weight> weightList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] weights = jsonArray.split(BRACKET_SLASH);
+            weightList.add(GSON.fromJson(weights[0], Weight.class));
+            for (int i = 1; i < weights.length; i++) {
+                weights[i] = BRACKET + weights[i];
+                weightList.add(GSON.fromJson(weights[i], Weight.class));
+            }
+        }
+        return weightList;
+    }
+
+    /**
+     * Updates the weight's value.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the weight was created
+     * @param value Value the weight will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateWeightField(String accessToken, String owner, String petName, DateTime date,
+                               Object value) throws ExecutionException, InterruptedException {
```

Comment:
If possible value should be of a specific class, not an object, in order to avoid values of invalid types. In this case value is always a double

Reply:
Solved!